src/components/ValueInput.vue

```diff
@@ -0,0 +1,48 @@
+<template>
+	<ItemValueLookup
+		v-if="datatype === 'wikibase-item'"
+		:value="value"
+		@input="$emit('input', $event)"
+		:error="error"
+		:disabled="disabled"
+	/>
+	<TextInput
+		v-else
+		:value="value"
+		:label="$i18n('query-builder-input-value-label')"
+		@input="$emit('input', $event)"
+		:error="error && { message: $i18n(error.message), type: error.type }"
+		:placeholder="$i18n('query-builder-input-value-placeholder')"
+		:disabled="disabled"
+	/>
```

Comment:
I really like the looks of this, thanks! I had looked at `<component is=""` previously and it didn't seem exactly right, but your approach is very convincing.

Then the first step for this would be to wrap the `TextInput` into its own component to align the interfaces, right?. That was on my Todo-list anyway, I will look into it.

Reply:
> Then the first step for this would be to wrap the TextInput into its own component to align the interfaces, right?

Yes, that sounds good!